File: javascript/2.js
function longestArithSeqLength(nums) {
   let maxLength = 0;

   for (let i = 0; i < nums.length; i++) {
       for (let y = i+1; y < nums.length; y++) {
           let diff = nums[y] - nums[i];
           let lastNum = nums[y];
           let count = 2;

           for (let k = y+1; k < nums.length; k++) {
               if (nums[k] - lastNum == diff) {
                  lastNum = nums[k];
                  count++;
               }
           }
           maxLength = Math.max(maxLength, count);
       }
   }

   return maxLength;
}

let nums = [3, 6, 9, 12, 1231412];
console.log("The longest Arithmetic Subsequence in `nums` is: " + longestArithSeqLength(nums));

let nums2 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11];
console.log("The longest Arithmetic Subsequence in `nums2` is: " + longestArithSeqLength(nums2));
 

File: python/2.py
def longest_arith_seq_length(nums):
    max_length = 0

    for i in range(len(nums)):
     for y in range(i+1, len(nums)):
         diff = nums[y] - nums[i]
         last_num = nums[y]
         count = 2
    
         for k in range(y+1, len(nums)):
             if nums[k] - last_num == diff:
                 last_num = nums[k]
                 count += 1
         max_length = max(max_length, count)
    
    return max_length

nums = [3, 6, 9, 12, 1231412]
print(f"The longest Arithmetic Subsequence in `nums` is: {longest_arith_seq_length(nums)}")

nums2 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
print(f"The longest Arithmetic Subsequence in `nums2` is: {longest_arith_seq_length(nums2)}")
